wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceRuntimes.java

```diff
@@ -130,7 +134,7 @@ public void validate(Environment environment)
       throw new NotFoundException("Infrastructure not found for type: " + type);
     }
 
-    infraByRecipe.get(type).estimate(environment);
+    environmentFactories.get(environment.getRecipe().getType()).create(environment);
```

Comment:
Potentially NPE can be thrown here. It's needed to check that the corresponding environment parser is present before using it.

Reply:
sure, fixed